### PR TITLE
Update 3rd-party components and fix CVE-2023-28840, CVE-2023-28841, C…

### PR DESCRIPTION
**Pull request by tdrozdovsky**
…VE-2023-28842

# Description

Update 3rd-party components and fix CVEs

Fixes #693 #694 #695 #696
Closes #671 #692 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] CI system update

**Test Configuration**:
* OS type & version: Ubuntu 20.04
* Hardware: x86-64 (e.g., x86, x86-64, arm, arm64)
* Toolchain: Docker v20.10 & Go v1.19
* Edge Orchestration Release: v1.2.0

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,7 +5,7 @@ go 1.19
 require (
 	github.com/casbin/casbin v1.9.1
 	github.com/docker/cli v20.10.17+incompatible
-	github.com/docker/docker v17.12.0-ce-rc1.0.20201201034508-7d75c1d40d88+incompatible
+	github.com/docker/docker v20.10.24+incompatible
```
Comment (tiokim) [inline go.mod:8]: Have you tested the basic functions of Home Edge?
 I'm a little worried about updating the docker version without updating source code of [containerexecutor](https://github.com/lf-edge/edge-home-orchestration-go/blob/master/internal/controller/servicemgr/executor/containerexecutor/containerexecutor.go).

Comment (tdrozdovsky) [reply go.mod:8]: I have started the "hello-world" container and everything works for me
```INFO[2023-06-07T16:17:01Z]containerexecutor.go:71 Execute [containerexecutor] hello-world docker run -v /var/run:/var/run:rw -e REQUEST=10.0.2.15 hello-world@sha256:fc6a51919cfeb2e6763f62b6d9e8815acbf7cd2e476ea353743570610737b752 
INFO[2023-06-07T16:17:01Z]containerexecutor.go:72 Execute [containerexecutor] parameter length : 7     
{"status":"Pulling from library/hello-world","id":"docker.io/library/hello-world@sha256:fc6a51919cfeb2e6763f62b6d9e8815acbf7cd2e476ea353743570610737b752"}
{"status":"Digest: sha256:fc6a51919cfeb2e6763f62b6d9e8815acbf7cd2e476ea353743570610737b752"}
{"status":"Status: Image is up to date for hello-world@sha256:fc6a51919cfeb2e6763f62b6d9e8815acbf7cd2e476ea353743570610737b752"}
INFO[2023-06-07T16:17:02Z]containerexecutor.go:92 Execute [containerexecutor] create container : 725ea50eb5 
2023-06-07T16:17:02.734269663Z 
2023-06-07T16:17:02.734300439Z Hello from Docker!
2023-06-07T16:17:02.734302886Z This message shows that your installation appears to be working correctly.
2023-06-07T16:17:02.734304687Z 
2023-06-07T16:17:02.734306407Z To generate this message, Docker took the following steps:
2023-06-07T16:17:02.734308099Z  1. The Docker client contacted the Docker daemon.
2023-06-07T16:17:02.734309827Z  2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
2023-06-07T16:17:02.734311637Z     (amd64)
2023-06-07T16:17:02.734313398Z  3. The Docker daemon created a new container from that image which runs the
2023-06-07T16:17:02.734315156Z     executable that produces the output you are currently reading.
2023-06-07T16:17:02.734316880Z  4. The Docker daemon streamed that output to the Docker client, which sent it
2023-06-07T16:17:02.734318692Z     to your terminal.
2023-06-07T16:17:02.734320410Z 
2023-06-07T16:17:02.734322018Z To try something more ambitious, you can run an Ubuntu container with:
2023-06-07T16:17:02.734323768Z  $ docker run -it ubuntu bash
2023-06-07T16:17:02.734325458Z 
2023-06-07T16:17:02.734327104Z Share images, automate workflows, and more with a free Docker ID:
2023-06-07T16:17:02.734328837Z  https://hub.docker.com/
2023-06-07T16:17:02.734330499Z 
2023-06-07T16:17:02.734332142Z For more examples and ideas, visit:
2023-06-07T16:17:02.734333870Z  https://docs.docker.com/get-started/
2023-06-07T16:17:02.734335670Z 
INFO[2023-06-07T16:17:02Z]containerexecutor.go:118 Execute [containerexecutor] container execution status : 0 
INFO[2023-06-07T16:17:02Z]orchestrationapi.go:474 listenNotify [orchestrationapi] service status changed [appNames:hello-world][status:Finished] 
INFO[2023-06-07T16:17:09Z]discovery.go:859 activeDiscovery [discoverymgr] activeDiscovery!!!            
INFO[2023-06-07T16:17:09Z]discovery.go:573 func1 [deviceDetectionRoutine] edge-orchestration-10163d4f-104c-46c5-9e2a-cf8d02b8acb6 
INFO[2023-06-07T16:17:09Z]discovery.go:574 func1 [deviceDetectionRoutine] confInfo    : ExecType(container), Platform(docker) 
INFO[2023-06-07T16:17:09Z]discovery.go:575 func1 [deviceDetectionRoutine] netInfo     : IPv4([10.0.2.15]), RTT(0) 
INFO[2023-06-07T16:17:09Z]discovery.go:576 func1 [deviceDetectionRoutine] serviceInfo : Services([DataStorage]) 
```

**Review by tiokim — APPROVED** [review]
LGTM

**Review by MoonkiHong — APPROVED** [review]
+1

**Pull request merged**